xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-macro/src/main/java/org/xwiki/icon/macro/DisplayIconMacroParameters.java

```diff
@@ -100,4 +102,24 @@ public void setFallback(boolean fallback)
     {
         this.fallback = fallback;
     }
+
+    /**
+     * @since 16.8.0RC1
+     * @return the text alternative picked for the icon
+     */
+    public String getTextAlternative()
+    {
+        return this.textAlternative;
+    }
+
+    /**
+     * @since 16.8.0RC1
+     * @param textAlternative a text alternative for the icon
+     */
+    @PropertyName("Text Alternative")
+    @PropertyDescription("A text alternative for this icon.")
+    public void setTextAlternative(String textAlternative)
+    {
+        this.textAlternative = textAlternative;
+    }
```

Comment:
Adding methods to a class isn't breaking anything, so I wouldn't expect any errors.

Reply:
Right, not an interface. Still missing the unstable annotations then.

Reply:
With the quality profile, I hit the ClassFanOutComplexity, working on fixing it...

Reply:
Added the annotation in https://github.com/xwiki/xwiki-platform/pull/3123/commits/7d12a733a916a085fa20cc901df749a98bd198fd :+1: 
Also reduced the fanout complexity by choosing to use a paragraph block instead of a FormatBlock for the text alternative. Now passes `mvn clean install -f xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-macro -Pquality` successfully.

Reply:
I'm sorry but using a paragraph block for the text alternative is not okay, you cannot nest a paragraph inside another paragraph and the icon macro needs to be usable in a paragraph.

Reply:
Reverted this change of block nature in https://github.com/xwiki/xwiki-platform/pull/3123/commits/51edb3b88c0baf8a30996e359252dcdbb89ef45d . 
Instead of this, I set some methods in their own class in order to reduce the ClassFanOutComplexity of DisplayIconMacro.

The PR with these new changes pass quality tests :)
`mvn clean install -f xwiki-platform-core/xwiki-platform-icon/xwiki-platform-icon-macro -Pquality`

Reply:
I went with a solution without components since those methods are quite specific and I don't think they'd need to be reused in another context.